File: mario.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    // Prompt user for height of the half-pyramid. Initialize height.
    int height;

    do
    {
        height = get_int("Enter the desired height (0 to 23): ");
    }
    while(height < 0 || height > 23);

    // Construct the pyramid of the specified height. Row loop below.
    for (int i = 0; i < height; i++)
    {
        // Print spaces to make the pyramid right aligned. Space loop below.
        for (int j = (height - (i+1)); j > 0; j--)
        {
            printf(" ");
        }

        // Print hashes. Hash loop below.
        if (i == 0)
        {
            printf("#  #");
        }
        else
        {
            // Left hashes.
            for (int j = 0; j < (i+1); j++)
            {
                printf("#");
            }

            // Print 2 spaces.
            printf("  ");

            // Right hashes.
            for (int j = 0; j < (i+1); j++)
            {
                printf("#");
            }
        }


        // Print a new line to go to the next row
        printf("\n");
    }
}

File: sentimental/mario.py
# mario.py

# Check if input is numeric and between 0 and 23
while True:
    try:
        height = int(input("Please enter the size of the pyramid: "))
    except ValueError:
        print("Invalid input! Please enter a number between 0 and 23.")
        continue
    if height < 0 or height > 23:
        print("Invalid input! Please enter a number between 0 and 23.")
    else:
        break


# Build the pyramid
for i in range(height):
    # Loop to add spaces for left-indenting
    for j in range(height - (i + 1)):
        # Print space
        print(" ", end="")

    # Print hashes
    if i == 0:
        print("#  #", end="")
    else:
        for k in range(i + 1):
            # Print left hashes
            print("#", end="")

        # Print 2 spaces
        print("  ", end="")

        # Print right hashes
        for k in range(i + 1):
            # Print left hashes
            print("#", end="")

    # Print a new line before the next loop
    print()